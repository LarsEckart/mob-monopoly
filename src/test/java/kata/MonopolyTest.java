package kata;

import com.github.larseckart.tcr.TestCommitRevertMainExtension;
import org.approvaltests.Approvals;
import org.approvaltests.StoryBoard;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

import static org.assertj.core.api.Assertions.assertThat;

@ExtendWith(TestCommitRevertMainExtension.class)
class MonopolyTest {

    @Test
    void testBasicPlay() {
        StoryBoard story = new StoryBoard();
        Monopoly monopoly = new Monopoly(3);
        story.addFrame(monopoly);
        story.addFrame(monopoly.move(5).buyProperty().endTurn());
        story.addFrame(monopoly.move(3).buyProperty().endTurn());
        story.addFrame(monopoly.move(3).endTurn());
        Approvals.verify(story);
    }

    @Test
    void testIncomeTaxAndChanceAndJail() {
        StoryBoard story = new StoryBoard();
        ChancePile.stack(new ChanceCard.BankCard());
        Monopoly monopoly = new Monopoly(3);
        story.addFrame(monopoly);
        story.addFrame(monopoly.move(10).endTurn());
        story.addFrame(monopoly.move(4).endTurn());
        story.addFrame(monopoly.move(7).endTurn());
        Approvals.verify(story);
    }

    @Test
    void testOtherChances() {
        StoryBoard story = new StoryBoard();
        ChancePile.stack(new ChanceCard.BankCard()
                , new ChanceCard.GoToStCharlesPlace()
                , new ChanceCard.GoBack3Spaces()
                , new ChanceCard.SpeedingFine()
                , new ChanceCard.ChairmanOfTheBoard()
                , new ChanceCard.GetOutOfJailFree()
        );
        int chanceCards = ChancePile.size();
        Monopoly monopoly = new Monopoly(chanceCards);
        story.addFrame(monopoly);
        for (int i = 0; i < chanceCards; i++) {
            if (i == 1) {
                story.addFrame(monopoly.move(7).buyProperty().endTurn());
            } else {
                story.addFrame(monopoly.move(7).endTurn());
            }
        }

        Approvals.verify(story);
    }

    @Test
    void testPlayerDetails() {
        Player player = new Player("Player 1");

        assertThat(player.details()).isEqualTo("Player 1 ($1500) ");
    }

    @Test
    void testPlayerLocation() {
        Player player = new Player("Player 1");

        assertThat(player.location()).isEqualTo(0);
    }

}
