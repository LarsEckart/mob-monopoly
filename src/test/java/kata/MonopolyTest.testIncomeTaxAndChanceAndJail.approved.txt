Initial:
|          GO          | [Player 1, Player 2, Player 3] |
| Mediterranean Avenue |               []               |
|   Community Chest    |               []               |
|    Baltic Avenue     |               []               |
|      Income Tax      |               []               |
|   Reading Railroad   |               []               |
|   Oriental Avenue    |               []               |
|        Chance        |               []               |
|    Vermont Avenue    |               []               |
|  Connecticut Avenue  |               []               |
|         Jail         |               []               |
|  St. Charles Place   |               []               |
Player 1 ($1500) 
Player 2 ($1500) 
Player 3 ($1500) 
current player: 1

Frame #1:
|          GO          | [Player 2, Player 3] |
| Mediterranean Avenue |          []          |
|   Community Chest    |          []          |
|    Baltic Avenue     |          []          |
|      Income Tax      |          []          |
|   Reading Railroad   |          []          |
|   Oriental Avenue    |          []          |
|        Chance        |          []          |
|    Vermont Avenue    |          []          |
|  Connecticut Avenue  |          []          |
|         Jail         |      [Player 1]      |
|  St. Charles Place   |          []          |
Player 1 ($1500) 
Player 2 ($1500) 
Player 3 ($1500) 
current player: 2

Frame #2:
|          GO          | [Player 3] |
| Mediterranean Avenue |     []     |
|   Community Chest    |     []     |
|    Baltic Avenue     |     []     |
|      Income Tax      | [Player 2] |
|   Reading Railroad   |     []     |
|   Oriental Avenue    |     []     |
|        Chance        |     []     |
|    Vermont Avenue    |     []     |
|  Connecticut Avenue  |     []     |
|         Jail         | [Player 1] |
|  St. Charles Place   |     []     |
Player 1 ($1500) 
Player 2 ($1300) 
Player 3 ($1500) 
current player: 3

Frame #3:
|          GO          |     []     |
| Mediterranean Avenue |     []     |
|   Community Chest    |     []     |
|    Baltic Avenue     |     []     |
|      Income Tax      | [Player 2] |
|   Reading Railroad   |     []     |
|   Oriental Avenue    |     []     |
|        Chance        | [Player 3] |
|    Vermont Avenue    |     []     |
|  Connecticut Avenue  |     []     |
|         Jail         | [Player 1] |
|  St. Charles Place   |     []     |
Player 1 ($1500) 
Player 2 ($1300) 
Player 3 ($1550) 
current player: 1